[Runtime] Auth args to compareTypeContextDescriptors.

nate-chandler opened the pull request:
When `swift_compareTypeContextDescriptors` was added, it did not auth the `TypeContextDescriptor` arguments that were passed to it.  Fix that here.

There are not any uses of this function yet, so there are no signs that need to be added.

nate-chandler commented:
@swift-ci please test

nate-chandler commented:
@swift-ci please test windows

jckarter commented:
Can we put the right `__ptrauth` modifier on the arguments directly?

nate-chandler commented:
@swift-ci please test

nate-chandler commented:
@swift-ci please test

jckarter reviewed (approved):
Thanks!

nate-chandler commented:
@swift-ci please test macos

nate-chandler commented:
@swift-ci please test windows

shahmishal commented:
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

nate-chandler commented:
@swift-ci please smoke test

nate-chandler commented:
@swift-ci please smoke test

nate-chandler commented:
@swift-ci please smoke test

nate-chandler commented:
@jckarter Unfortunately, it seems that the `__ptrauth` modifier cannot be applied directly to arguments:

```
include/swift/Runtime/Metadata.h|309 col 78| error: parameter types may not be qualified with __ptrauth; type is 'const TargetSignedContextPointer<swift::InProcess, TargetTypeContextDescriptor>' (aka 'swift::TargetTypeContextDescriptor<swift::InProcess> *const __ptrauth(2,1,44678)')
    const TargetSignedContextPointer<InProcess, TargetTypeContextDescriptor> lhs,
                                                                             ^
```

nate-chandler committed:
[Runtime] Auth args to compareTypeContextDescriptors.

When swift_compareTypeContextDescriptors was added, it did not auth the
TypeContextDesriptor arguments that were passed to it.  Fix that here.

There are not any uses of this function yet, so there are no signs that
need to be added.

nate-chandler commented:
@swift-ci please test
